new README (#767)
oneapi-src/oneDAL: README.md

<!--
******************************************************************************
* Copyright 2014-2020 Intel Corporation
  *
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
  *
* http://www.apache.org/licenses/LICENSE-2.0
  *
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
  *******************************************************************************/-->

# Intel&reg; oneAPI Data Analytics Library

> [Installation](#installation)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Documentation](http://intel.github.io/daal/)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Examples](#examples)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Get Help](https://software.intel.com/en-us/forums/intel-data-analytics-acceleration-library)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[How to Contribute](CONTRIBUTING.md)&nbsp;&nbsp;&nbsp;

Reviewer: Please update the old links, e.g. `http://intel.github.io/daal/` -> `http://oneapi-src.github.io/oneDAL/`

[![Build Status](https://dev.azure.com/daal/DAAL/_apis/build/status/intel.daal?branchName=master)](https://dev.azure.com/daal/DAAL/_build/latest?definitionId=3&branchName=master) ![License](https://img.shields.io/github/license/intel/daal.svg)

 Intel&reg; oneAPI Data Analytics Library (oneDAL) is a powerful machine liearning library that helps speed up big data analysis.
 Intel&reg; oneDAL solvers are also used in [Intel Distribution for Python]([https://software.intel.com/en-us/distribution-for-python](https://software.intel.com/en-us/distribution-for-python)) in scikit-learn optimization.

> ## Build yours high-performance data science application with intel&reg; DAAL

Reviewer 2: I think we should use oneDAL across readme

Reviewer: This also applies to images.

Author: Corrected in text and in charts

We provide highly optimized algorithmic building blocks for all stages of data analytics: **preprocessing**, **transformation**, **analysis**, **modeling**, **validation**, and **decision making**. Our algorithms suppost **batch**, **online**, and **distributed** processing modes of computation. 
## Build yours high-performance data science application with intel&reg; DAAL

 intel&reg; DAAL use all capabilities of your hardware, which allows you to get an incredible performance boost on the classic machine learning algorithms.
We provide highly optimized algorithmic building blocks for all stages of data analytics: **preprocessing**, **transformation**, **analysis**, **modeling**, **validation**, and **decision making**.

The current version of oneDAL provides Data Parallel C++ (DPC++) API extensions to the traditional C++ interface.

## Table of Contents <!-- omit in toc -->
Check out our [examples](#examples)  and [documentation](#documentation)  for information about our API

- [Technical Preview Features](#preview-features)
- [oneDAL and Intel(R) DAAL](#onedal-and-intelr-daal)
- [Installation](#installation)
- [Examples](#examples)
- [Documentation](#documentation)

> - [API](#api)

Reviewer: why did you remove TOC?

Author: I returned TOC and updated it.

> <img style="display:inline;" height=auto width=auto src="https://github.com/PivovarA/daal/raw/dev/apivovar-newReadme/.github/charts/IDP%20scikit-learn%20accelearation%20compared%20with%20stock%20scikit-learn.png"></a>

Reviewer 2: Guess it should be a way to refer images relatively so it will not be required to specify repo name

```suggestion
<img style="display:inline;" height=auto width=auto src=".github/charts/IDP%20scikit-learn%20accelearation%20compared%20with%20stock%20scikit-learn.png"></a>
```

Author: Made all links relative

> ![](https://github.com/PivovarA/daal/raw/dev/apivovar-newReadme/.github/charts/Intel%20DAAL%20KMeans%20strong%20scaling.png)  |  ![](https://github.com/PivovarA/daal/raw/dev/apivovar-newReadme/.github/charts/intel%20DAAL%20KMeans%20weak%20scaling.png)

Reviewer 3: For all images you are adding: technical details are required like hardware specs, dataset details, software versions. Or, at least, that results are based on Intel internal measurements.

Author: Yes, thanks for pointing that out. I have added descriptions to the charts with HW, SW and other details.

> ## Python API

Reviewer 3: Python code in one repo, Python API description & example in another confuses a bit. Probably, need to add some intro or something to avoid confusion.

Reviewer 3: No answer yet.

Author: I added an explanation to the Python API paragraph: `Intel(R) oneDAL has a python API that is provided as a standalone python library called daal4py`

                           fptype = "float",
                           method = "randomDense")

centroids = init_alg.compute(data).centroids
alg = d4p.kmeans(nClusters = 10, maxIterations = 50, fptype = "float",
                 accuracyThreshold = 0, assignFlag = False)
result = alg.compute(data, centroids)
```

Often data scientists require different tools for analysis regular and big data. daal4py offers various processing models, which makes it easy to enable distributed multi-node mode.
```python
import numpy as np
import pandas as pd
import daal4py as d4p

d4p.daalinit() # <-- Initialize SPMD mode
data = pd.read_csv("local_kmeans_data.csv", dtype = np.float32)

init_alg = d4p.kmeans_init(nClusters = 10,
                           fptype = "float",
                           method = "randomDense",
                           distributed = True) # <-- change model to distributed

centroids = init_alg.compute(data).centroids

alg = d4p.kmeans(nClusters = 10, maxIterations = 50, fptype = "float",
                 accuracyThreshold = 0, assignFlag = False,
                 distributed = True)  # <-- change model to distributed

result = alg.compute(data, centroids)
```

> ## Scikit-learn patching

Reviewer 3: The context is not aligned with the title

Reviewer 3: Not addressed yet.

Author: I moved scikit-learn patching paragraph to Python API paragraph and added explanation: `Python interface to efficient Intel® oneDAL provided by daal4py allows one to create scikit-learn compatible estimators, transformers, clusterers, etc. powered by oneDAL which are nearly as efficient as native programs.`

daal4py also have an API which matches API from scikit-learn.
This framework allows you to speed up your existing projects by changing one line of code

```python
from daal4py.sklearn.svm import SVC
from sklearn.datasets import load_digits

digits = load_digits()
X, y = digits.data, digits.target

svm = SVC(kernel='rbf', gamma='scale', C = 0.5).fit(X, y)
print(svm.score(X, y))
```

In addition daal4py provides an option to replace some scikit-learn methods by daal solvers which makes it possible to get a performance gain **without any code changes**. This approach is the basis of Intel distribution for python scikit-learn. You can patch stock scikit-learn by using the only following commandline flag
```bash
python -m daal4py my_application.py
```
Patches can also be enabled programmatically:
```python
from sklearn.svm import SVC
from sklearn.datasets import load_digits

> 

Reviewer 4: ```suggestion
from time import time
```

Author: Already fixed

svm_sklearn = SVC(kernel="rbf", gamma="scale", C=0.5)

> 

Reviewer 4: ```suggestion
digits = load_digits()
X, y = digits.data, digits.target
```

Author: Already fixed

start = time()
svm_sklearn = svm_sklearn.fit(X, y)
end = time()

> print(start-end) # output: 0.141261...

Reviewer 4: ```suggestion
print(end - start) # output: 0.141261...
```

Author: Already fixed

print(svm_sklearn.score(X, y)) # output: 0.9905397885364496

from daal4py.sklearn import patch_sklearn
patch_sklearn() # <-- apply patch
from sklearn.svm import SVC

svm_d4p = SVC(kernel="rbf", gamma="scale", C=0.5)

start = time()
svm_d4p = svm_d4p.fit(X, y)
end = time()

> print(start-end) # output: 0.032536...

Reviewer 4: ```suggestion
print(end - start) # output: 0.032536...
```

Author: Already fixed

print(svm_d4p.score(X, y)) # output: 0.9905397885364496
```

For more details browse our [daal4py documentation](https://intelpython.github.io/daal4py/).

## DAAL Samples

<img align="right" style="display:inline;" height=300 width=550 src="https://github.com/PivovarA/daal/raw/dev/apivovar-newReadme/.github/charts/intel%20DAAL%20Spark%20samples%20vs%20Apache%20Spark%20MLlib.png"></a>

Samples is a non official part of a project, that contains examples of how DAAL can be used in different applications.

**DAAL Spark Samples is a good example.**

> DAAL provides scala / java interfaces that match Apache Spark MlLib API and use DAAL solvers under the hood. This implementation allows you to get a 3-18X increase in performance compared to default Apache Spark MLlib.

Reviewer 3: Some footnote is required for such a performance claim: hardware specs, datasets, software versions, etc.

Author: Yes, thanks for pointing that out. I have added descriptions to the charts with HW, SW and other details.

  ## Examples

Except C++ and Python API DAAL also provide API for C++ SYCL and Java languages. Check out tabs below for more examples.
- [C++](https://github.com/intel/daal/tree/master/examples/daal/cpp)
- [C++ SYCL*](https://github.com/intel/daal/tree/master/examples/daal/cpp_sycl)
- [Java](https://github.com/intel/daal/tree/master/examples/daal/java)

> 

Reviewer 4: Why didn't you add examples for python?
```suggestion
- [Python](https://github.com/IntelPython/daal4py/tree/master/examples)
```

Author: Thanks. Added Python examples.

  ## Examples

Except C++ and Python API DAAL also provide API for C++ SYCL and Java languages. Check out tabs below for more examples.
- [C++](https://github.com/intel/daal/tree/master/examples/daal/cpp)
- [C++ SYCL*](https://github.com/intel/daal/tree/master/examples/daal/cpp_sycl)
- [Java](https://github.com/intel/daal/tree/master/examples/daal/java)

Data Examples for different computation modes:

- [Batch](https://github.com/oneapi-src/oneDAL/tree/master/examples/data/batch)
- [Distributed](https://github.com/oneapi-src/oneDAL/tree/master/examples/data/distributed)
- [Online](https://github.com/oneapi-src/oneDAL/tree/master/examples/data/online)
- [Batch](https://github.com/intel/daal/tree/master/examples/daal/data/batch)
- [Distributed](https://github.com/intel/daal/tree/master/examples/daal/data/distributed)
- [Online](https://github.com/intel/daal/tree/master/examples/daal/data/online)

## Documentation

- [Get Started](http://oneapi-src.github.io/oneDAL/getstarted.html)

> - [System Requirements](https://software.intel.com/en-us/articles/system-requirements-for-oneapi-data-analytics-library#)

Reviewer: why did you remove this?

updated link to system requirements:
```
- [System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)
```

Author: Thanks. I updated this item

> - [Get Started](http://intel.github.io/daal/getstarted.html)

Reviewer: ```suggestion
- [Get Started](http://oneapi-src.github.io/oneDAL/getstarted.html)
```

> - [oneDAL documentation](http://intel.github.io/daal/)

Reviewer: ```suggestion
- [oneDAL documentation](http://oneapi-src.github.io/oneDAL/)
```

> - [Specifications](https://spec.oneapi.com/oneDAL/index.html)

Reviewer: ```suggestion
- [Specifications](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/index.html)
```

> - [Release Notes](https://software.intel.com/en-us/articles/oneapi-dal-release-notes)

Reviewer: ```suggestion
- [Release Notes](https://software.intel.com/content/www/us/en/develop/articles/oneapi-dal-release-notes.html)
```

- [oneDAL documentation](http://oneapi-src.github.io/oneDAL/)
- [Specifications](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/index.html)
- [Get Started](http://intel.github.io/daal/getstarted.html)
- [oneDAL documentation](http://intel.github.io/daal/)
- [Specifications](https://spec.oneapi.com/oneDAL/index.html)
- [Release Notes](https://software.intel.com/en-us/articles/oneapi-dal-release-notes)

> - [Known Issues](https://oneapi-src.github.io/oneDAL/notes/known_issues.html)

Reviewer: why did you remove this?

Author: I returned this item

> - from [oneDAL home page](https://software.intel.com/en-us/oneapi/onedal) as a part of Intel&reg; oneAPI Base Toolkit.
> - from [GitHub\*](https://github.com/intel/daal/releases).

Reviewer: ```suggestion
- from [oneDAL home page](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onedal.html) as a part of Intel(R) oneAPI Base Toolkit.
- from [GitHub\*](https://github.com/oneapi-src/oneDAL/releases).
```

> ## Installation from Source

Reviewer: I'd either make it a level three heading, or remove it completely

> Technical preview features are introduced to gain early feedback from developers. A preview feature is subject to change in the future releases. Using a preview feature in a production code base is therefore strongly discouraged.

Reviewer: ```suggestion
Technical preview features are introduced to gain early feedback from developers. A technical preview feature is subject to change in the future releases. Using a technical preview feature in a production code base is therefore strongly discouraged.
```

After that line, please also add the following text proposed by @averbukh: 

```

In C++ APIs, technical preview features are located in `daal::preview` and `onedal::preview` namespaces. In Java APIs, technical preview features are located in packages that have the `com.intel.daal.preview` name prefix.

```

Author: Added

> ## oneDAL and Intel&reg; DAAL

Reviewer 3: Important info - I propose to have it closer to the top of the page, as it was before your changes.

Reviewer: I agree

Reviewer 2: +1

Author: I added this information after `Build yours high-performance data science application with intel(R) oneDAL paragraph.`

> |oneDAL       |2021.1-beta06|[master](https://github.com/oneapi-src/oneDAL)</br>[rls/onedal-beta06-rls](https://github.com/oneapi-src/oneDAL/tree/rls/onedal-beta06-rls)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/en-us/oneapi/onedal)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Documentation](http://oneapi-src.github.io/oneDAL/)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/en-us/articles/system-requirements-for-oneapi-data-analytics-library#)|
> |Intel&reg; DAAL|2020 Gold|[rls/daal-2020-rls](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-rls)</br>[rls/daal-2020-mnt](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-mnt) (contains ongoing fixes)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/en-us/daal)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Developer Guide](https://software.intel.com/en-us/daal-programming-guide)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/en-us/articles/intel-data-analytics-acceleration-library-2020-system-requirements)|

Reviewer: ```suggestion
|oneDAL       |2021.1-beta08|[master](https://github.com/oneapi-src/oneDAL)</br>[rls/onedal-beta08-rls](https://github.com/oneapi-src/oneDAL/tree/rls/onedal-beta08-rls)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onedal.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Documentation](http://oneapi-src.github.io/oneDAL/)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)|
|Intel(R) DAAL|2020 Gold|[rls/daal-2020-rls](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-rls)</br>[rls/daal-2020-mnt](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-mnt) (contains ongoing fixes)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/data-analytics-acceleration-library.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Developer Guide](https://software.intel.com/content/www/us/en/develop/documentation/daal-programming-guide/top.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/intel-data-analytics-acceleration-library-2020-system-requirements.html)|
```

Author: Thanks. I added this suggestion

<!--
******************************************************************************
* Copyright 2014-2020 Intel Corporation

>   *

Reviewer 3: New alignment looks strange - previous one looks better

Author: Thanks. I fixed it

* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
  *
* http://www.apache.org/licenses/LICENSE-2.0

>   *

Reviewer 3: New alignment looks strange - previous one looks better

* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.

>   *******************************************************************************/-->

Reviewer 3: New alignment looks strange - previous one looks better

Author: Thanks. I fixed it.

 Intel&reg; oneAPI Data Analytics Library (oneDAL) is a powerful machine liearning library that helps speed up big data analysis.
 Intel&reg; oneDAL solvers are also used in [Intel Distribution for Python]([https://software.intel.com/en-us/distribution-for-python](https://software.intel.com/en-us/distribution-for-python)) in scikit-learn optimization.

## Table of Contents
- [Build yours high-performance data science application with intel&reg; oneDAL](#build-yours-high-performance-data-science-application-with-intel-onedal)
- [Improve performance of your ML application](#improve-performance-of-your-ml-application)
- [Installation](#installation)
- [Installation from Source](#installation-from-source)
- [Technical Preview Features](#technical-preview-features)
- [Python API](#python-api)
- [Scikit-learn patching](#scikit-learn-patching)
- [oneDAL Samples](#onedal-samples)
- [Examples](#examples)
- [Documentation](#documentation)
- [oneDAL and Intel&reg; DAAL](#onedal-and-intel-daal)

## Build yours high-performance data science application with intel&reg; oneDAL

>  intel&reg; oneDAL use all capabilities of your hardware, which allows you to get an incredible performance boost on the classic machine learning algorithms.

Reviewer 3: New alignment looks strange - previous one looks better

Author: Thanks. I fixed it.

>  intel&reg; oneDAL use all capabilities of your hardware, which allows you to get an incredible performance boost on the classic machine learning algorithms.

Reviewer 3: your hardware -> the hardware or Intel(R) hardware

Author: replace your hardware to intel(R) hardware

>  intel&reg; oneDAL use all capabilities of your hardware, which allows you to get an incredible performance boost on the classic machine learning algorithms.

Reviewer 3: oneDAL use -> oneDAL uses

Author: replace use to uses

>  intel&reg; oneDAL use all capabilities of your hardware, which allows you to get an incredible performance boost on the classic machine learning algorithms.

Reviewer 3: "an incredible performance boost" - I suggest to specify in more clear way, because now it sounds ambiguous due to the indefinite article

We provide highly optimized algorithmic building blocks for all stages of data analytics: **preprocessing**, **transformation**, **analysis**, **modeling**, **validation**, and **decision making**.

The current version of oneDAL provides Data Parallel C++ (DPC++) API extensions to the traditional C++ interface.

Check out our [examples](#examples)  and [documentation](#documentation)  for information about our API

## Improve performance of your ML application

> oneDAL is the perfect solution if you are interested in a high-performance machine learning framework. oneDAL uses machine learning algorithms optimizations and full power of your hardware to achieve greater performance than alternative solutions offers.

Reviewer 3: your hardware -> Intel(R) hardware

The current version of oneDAL provides Data Parallel C++ (DPC++) API extensions to the traditional C++ interface.

## Table of Contents <!-- omit in toc -->
Check out our [examples](#examples)  and [documentation](#documentation)  for information about our API

- [Technical Preview Features](#preview-features)
- [oneDAL and Intel(R) DAAL](#onedal-and-intelr-daal)
- [Installation](#installation)
- [Examples](#examples)
- [Documentation](#documentation)
- [API](#api)
## Improve performance of your ML application

oneDAL is the perfect solution if you are interested in a high-performance machine learning framework. oneDAL uses machine learning algorithms optimizations and full power of your hardware to achieve greater performance than alternative solutions offers.

## Technical Preview Features 
 
Technical preview features are introduced to gain early feedback from developers. A preview feature is subject to change in the future releases. Using a preview feature in a production code base is therefore strongly discouraged.

> The only preview feature at the moment is `MultiNodeBatch` for K-Means, a stepless distributed algorithm based on oneCCL.

Reviewer 3: Why section about Technical Preview Features was moved?

Author: I added this information after Build yours high-performance data science application with intel(R) oneDAL paragraph.

> Besides superior performance on a single node, the distribution mechanics of oneDAL provides excellent strong and weak scaling.

Reviewer 3: Inconsistent term "distribution mechanics" with another docs

Author: Thanks. I replaced it with `the oneDAL distributed multi-node algorithms implementations`

Reviewer 3: We normally use 'distributed computation mode'

Author: Thanks for pointing that out.  Replaced it with distributed computation mode

> Besides superior performance on a single node, the distribution mechanics of oneDAL provides excellent strong and weak scaling.

Reviewer 3: "excellent strong and weak scaling" - Can we have reference to some visual context here, which can be considered as proof.

Author: There are 2 charts below this sentence that illustrate this. I added a tag: `check charts below`

The only preview feature at the moment is `MultiNodeBatch` for K-Means, a stepless distributed algorithm based on oneCCL.

## Python API

Intel&reg; oneDAL has a python API that is provided as a standalone python library called [daal4py](https://github.com/IntelPython/daal4py).
Below is an example of how daal4py can be used for calculation KMeans clusters

```python
import numpy as np
import pandas as pd
import daal4py as d4p

data = pd.read_csv("local_kmeans_data.csv", dtype = np.float32)

init_alg = d4p.kmeans_init(nClusters = 10,
                           fptype = "float",
                           method = "randomDense")

centroids = init_alg.compute(data).centroids
alg = d4p.kmeans(nClusters = 10, maxIterations = 50, fptype = "float",
                 accuracyThreshold = 0, assignFlag = False)
result = alg.compute(data, centroids)
```

Often data scientists require different tools for analysis regular and big data. daal4py offers various processing models, which makes it easy to enable distributed multi-node mode.

> ```python

Reviewer 2: Think it worth it to add examples with GPU support and DPPL - @rlnx  @michael-smirnov   - any candidates for adding here?

Author: Good idea. It could be added as a new paragraph in Python API section

import numpy as np
import pandas as pd
import daal4py as d4p

d4p.daalinit() # <-- Initialize SPMD mode
data = pd.read_csv("local_kmeans_data.csv", dtype = np.float32)

init_alg = d4p.kmeans_init(nClusters = 10,
                           fptype = "float",
                           method = "randomDense",
                           distributed = True) # <-- change model to distributed

centroids = init_alg.compute(data).centroids

alg = d4p.kmeans(nClusters = 10, maxIterations = 50, fptype = "float",
                 accuracyThreshold = 0, assignFlag = False,
                 distributed = True)  # <-- change model to distributed

result = alg.compute(data, centroids)
```

## Scikit-learn patching

> Python interface to efficient Intel® Data Analytics and Acceleration Library (DAAL) provided by daal4py allows one to create scikit-learn compatible estimators, transformers, clusterers, etc. powered by DAAL which are nearly as efficient as native programs.

Reviewer 2: Old DAL name used

Author: Thanks. Replace DAAL to oneDAL

```bash
python -m daal4py my_application.py
```
Patches can also be enabled programmatically:
```python
from sklearn.svm import SVC
from sklearn.datasets import load_digits
from time import time

svm_sklearn = SVC(kernel="rbf", gamma="scale", C=0.5)

digits = load_digits()
X, y = digits.data, digits.target

start = time()
svm_sklearn = svm_sklearn.fit(X, y)
end = time()
print(end - start) # output: 0.141261...
print(svm_sklearn.score(X, y)) # output: 0.9905397885364496

from daal4py.sklearn import patch_sklearn
patch_sklearn() # <-- apply patch
from sklearn.svm import SVC

svm_d4p = SVC(kernel="rbf", gamma="scale", C=0.5)

start = time()
svm_d4p = svm_d4p.fit(X, y)
end = time()
print(end - start) # output: 0.032536...
print(svm_d4p.score(X, y)) # output: 0.9905397885364496
```

For more details browse our [daal4py documentation](https://intelpython.github.io/daal4py/).

## oneDAL Samples

<img align="right" style="display:inline;" height=300 width=550 src="docs/readme-charts/intel%20oneDAL%20Spark%20samples%20vs%20Apache%20Spark%20MLlib.png"></a>

> Samples is a non official part of a project, that contains examples of how oneDAL can be used in different applications.

Reviewer 2: i think samples are part of project?

**oneDAL Spark Samples is a good example.**
oneDAL provides scala / java interfaces that match Apache Spark MlLib API and use oneDAL solvers under the hood. This implementation allows you to get a 3-18X increase in performance compared to default Apache Spark MLlib.

  ## Examples

Except C++ and Python API oneDAL also provide API for C++ SYCL and Java languages. Check out tabs below for more examples.
- [C++](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/cpp)

> - [C++ SYCL\*](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/cpp_sycl)

Reviewer 2: Think we should include new C++ APIs ?  And also paths are obsolete

Author: Added

- [Java](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java)
- [Python](https://github.com/IntelPython/daal4py/tree/master/examples)

Data Examples for different computation modes:

- [Batch](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/data/batch)
- [Distributed](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/data/distributed)

> - [Online](https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/data/online)

Reviewer 2: Not sure we should reference only DAAL examples

Author: It seems to me that keeping data references is not a good idea. We have a description of the computation modes in the paragraphs above, as well as an example of how to run distributed algorithms with daal4py

## Documentation
- [Get Started](http://oneapi-src.github.io/oneDAL/getstarted.html)
- [System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)
- [oneDAL documentation](http://oneapi-src.github.io/oneDAL/)

> - [Specifications](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/index.html)

Reviewer 2: ```suggestion
- [Specification](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/index.html)
```

Author: Thanks. I fixed this

- [Release Notes](https://software.intel.com/content/www/us/en/develop/articles/oneapi-dal-release-notes.html)
- [Known Issues](https://oneapi-src.github.io/oneDAL/notes/known_issues.html)

## oneDAL and Intel&reg; DAAL

Intel&reg; oneAPI Data Analytics Library is an extenstion of Intel&reg; Data Analytics Acceleration Library (Intel&reg; DAAL).

> This repository contains branches corresponding to both oneAPI and classical versions of the library. We encourage you to use oneDAL located under the `master` branch.

Reviewer 2: I think we also should add mention on internal split even within single branch - like oneDAL have old APIs and newAPIs

Reviewer 2: This can be handled in separate PRs

|Product|Latest release|Branch|Resources|
|-------|--------------|------|:-------------:|
|oneDAL       |2021.1-beta08|[master](https://github.com/oneapi-src/oneDAL)</br>[rls/onedal-beta08-rls](https://github.com/oneapi-src/oneDAL/tree/rls/onedal-beta08-rls)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onedal.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Documentation](http://oneapi-src.github.io/oneDAL/)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)|

> |Intel&reg; DAAL|2020 Gold|[rls/daal-2020-rls](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-rls)</br>[rls/daal-2020-mnt](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-mnt) (contains ongoing fixes)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/data-analytics-acceleration-library.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Developer Guide](https://software.intel.com/content/www/us/en/develop/documentation/daal-programming-guide/top.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/intel-data-analytics-acceleration-library-2020-system-requirements.html)|

Reviewer 2: Shouldn't this be U2 branch?

Reviewer 2: Links point to mnt branch but names are targeting 2020 Gold.

* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*

> * http://www.apache.org/licenses/LICENSE-2.0

Reviewer 5: ?

* limitations under the License.
*******************************************************************************/-->

# Intel&reg; oneAPI Data Analytics Library

[Installation](#installation)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Documentation](http://intel.github.io/daal/)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Examples](#examples)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[Get Help](https://software.intel.com/en-us/forums/intel-data-analytics-acceleration-library)&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;[How to Contribute](CONTRIBUTING.md)&nbsp;&nbsp;&nbsp;

> [![Build Status](https://dev.azure.com/daal/DAAL/_apis/build/status/intel.daal?branchName=master)](https://dev.azure.com/daal/DAAL/_build/latest?definitionId=3&branchName=master) ![License](https://img.shields.io/github/license/intel/daal.svg)

Reviewer 2: This should be returned back to oneapi-src  

![image](https://user-images.githubusercontent.com/18015221/91384663-d6a1e180-e837-11ea-91e1-99644fcaa2f7.png)

Intel&reg; oneAPI Data Analytics Library (oneDAL) is a powerful machine liearning library that helps speed up big data analysis.

Intel&reg; oneAPI Data Analytics Library is an extenstion of Intel&reg; Data Analytics Acceleration Library (Intel&reg; DAAL).

> Intel&reg; oneDAL solvers are also used in [Intel Distribution for Python]([https://software.intel.com/en-us/distribution-for-python](https://software.intel.com/en-us/distribution-for-python)) in scikit-learn optimization.

Reviewer 5: 3 sentences which start with "Intel&reg; oneAPI Data Analytics Library" look not so good. Better to write:

"
Intel® oneAPI Data Analytics Library (oneDAL) is a powerful machine liearning library that helps speed up big data analysis. oneDAL solvers are used in Intel Distribution for Python in Scikit-learn optimization.

Intel® oneDAL is an extension of Intel® Data Analytics Acceleration Library (Intel® DAAL).
"

P.S. also an error in a word "extenstion "

## Table of Contents
- [Build yours high-performance data science application with intel&reg; oneDAL](#build-yours-high-performance-data-science-application-with-intel-onedal)

> - [Technical Preview Features](#technical-preview-features)

Reviewer 5: I prefer to move this before point "oneDAL and Intel® DAAL"

- [Python API](#python-api)
- [Scikit-learn patching](#scikit-learn-patching)
- [Distributed multi-node mode](#distributed-multi-node-mode)
- [Installation](#installation)
- [Installation from Source](#installation-from-source)
- [Examples](#examples)
- [Samples](#samples)

> - [oneDAL Apache Spark MLlib samples](#onedal-apache-spark-mllib-samples)

Reviewer 5: Move acceleration of MLLib after "Distributed multi-node mode" and just add reference to "Samples" part

- [System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)
- [oneDAL documentation](http://oneapi-src.github.io/oneDAL/)
- [Specification](https://spec.oneapi.com/versions/latest/elements/oneDAL/source/index.html)
- [Release Notes](https://software.intel.com/content/www/us/en/develop/articles/oneapi-dal-release-notes.html)
- [Known Issues](https://oneapi-src.github.io/oneDAL/notes/known_issues.html)

## Samples
Samples is an examples of how oneDAL can be used in different applications.
- [Apache Arrow](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/arrow)
- [KDB](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/kdb)
- [MPI](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mpi)
- [MySQL](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mysql)
- [oneCCL](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/oneccl)
- [Hadoop](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/java/hadoop)
- [Java Spark](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/java/spark)

> - [Scala Spark](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/scala/spark)

Reviewer 5: I think we need to extend description of the samples. **But we can to it in separate PR.**
